lib/DMV_setup.rb

```diff
@@ -0,0 +1,50 @@
+class DMVSetup
+    attr_reader :facilities
+
+    def initialize
+        @facilities = []
+    end
+
+    def create_facilities(facility_data)
+        facility_data.each do |facility|
+            filtered_data = filter_data(facility)
+            @facilities << Facility.new(filtered_data)
+        end
+    end
+
+    def filter_data(data)
+        filtered_set = Hash.new("")
+        address_pair = {}
+        data.each_pair do |key, value|
+          if (key.to_s.include?("office") || key.to_s.include?("name")) && !(key.to_s.include?("type")) && !(key.to_s.include?("info"))
+            filtered_set[:name] = value
+          elsif key.to_s.include?("phone") && !(key.to_s.include?('text'))
+            filtered_set[:phone] = value
+          elsif key.to_s.include?("address") || key.to_s.include?("city") || key.to_s.include?("state") || key.to_s.include?("zip") 
+            address_pair[(key.to_s)] = value
+          end
+        end
+        address = address_config(address_pair)
+        filtered_set[:address] = address
+        filtered_set
+    end
+
+    def address_config(data)
+        address_string = []
+        data.each_pair do |key, value|
+            if key.include?("city")
+                address_string[2] = value
+            elsif key.include?("state") 
+                address_string[3] = value
+            elsif key.include?("zip")
+                address_string[4] = value
+            elsif key.include?("address") && value.include?("Suite")
+                address_string[1] = value
+            elsif key.include?("address") && !(key.include?("id")) && !(value.include?("Suite"))
+                address_string[0] = value
+            end
+        end
+        address_string = address_string.compact
+        address_string.join(" ")
+    end
```

Comment:
Really great use of helper methods here to adhere to single repsonsibility for your methods.